### PR TITLE
user/bupstash: new package

### PR DESCRIPTION
**Pull request by tjheeta**

### CODE REVIEW

File: user/bupstash/template.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+maintainer = "tj <tjheeta@gmail.com>"
+license = "MIT"
+url = "https://github.com/andrewchambers/bupstash"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "a2ce4eeb2caa881a778e823cc70d39c2adb0b301f737c55e44a4dd0fbd6a4265"
+
+def post_patch(self):
+    from cbuild.util import cargo
+
+    # Adjust bupstash dependencies for musl changes of -D_LARGEFILE_SOURCE / -D_LARGEFILE64_SOURCE
```
Comment (nekopsykose) [inline user/bupstash/template.py:18]: you should update the lockfile to update the libc crate version instead in a patch (and use prepare_after_patch=True). see alpine for instance

**Comment by nekopsykose** [comment]
apply https://github.com/andrewchambers/bupstash/pull/381 too

File: user/bupstash/template.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+pkgname = "bupstash"
+pkgver = "0.12.0"
+pkgrel = 0
+build_style = "cargo"
+make_build_args = []
+hostmakedepends = ["cargo-auditable", "pkgconf"]
+makedepends = ["libsodium-devel"]
+pkgdesc = "Bupstash is a tool for encrypted backups"
```
Comment (nekopsykose) [inline user/bupstash/template.py:8]:
```suggestion
pkgdesc = "Tool for encrypted backups"
```

File: user/bupstash/template.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+pkgname = "bupstash"
+pkgver = "0.12.0"
+pkgrel = 0
+build_style = "cargo"
+make_build_args = []
```
Comment (triallax) [inline user/bupstash/template.py:5]:
```suggestion
```

File: user/bupstash/template.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+hostmakedepends = ["cargo-auditable", "pkgconf"]
+makedepends = ["libsodium-devel"]
+pkgdesc = "Bupstash is a tool for encrypted backups"
+maintainer = "tj <tjheeta@gmail.com>"
+license = "MIT"
+url = "https://github.com/andrewchambers/bupstash"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
```
Comment (triallax) [inline user/bupstash/template.py:12]:
```suggestion
url = "https://bupstash.io"
source = f"https://github.com/andrewchambers/bupstash/archive/refs/tags/v{pkgver}.tar.gz"
```

**Comment by q66** [comment]
drafted for a long time with no changes, feel free to reopen if you want

**Pull request closed without merging**